### PR TITLE
style: simplify some statements for readability

### LINKED ISSUE #1961: Some statements can be simplified for readability
We can simplify some statements. For example, we can replace

```rust
&format!("__pgrx_internals_aggregate_{}", snake_case_target_ident),
```

with

```rust
&format!("__pgrx_internals_aggregate_{snake_case_target_ident}"),
```

which is a bit more readable, and should improve maintainability.

### PR DESCRIPTION
**Pull request by hamirmahal**
Closes #1961

### CODE REVIEW

**Review by hamirmahal — COMMENTED** [review]
I updated `rustup` to match the version in CI. This should fix failing formatting checks.

File: pgrx/src/datum/unbox.rs [hunk 1]
```diff
@@ -122,7 +122,10 @@ unsafe impl UnboxDatum for str {
 }
 
 unsafe impl UnboxDatum for &str {
-    type As<'src> = &'src str where Self: 'src;
+    type As<'src>
+        = &'src str
+    where
+        Self: 'src;
```
Comment (workingjubilee) [inline pgrx/src/datum/unbox.rs]: ...I would prefer not to?

Comment (hamirmahal) [reply pgrx/src/datum/unbox.rs]: That's how the version of `rustup` in CI formats things.

Comment (workingjubilee) [reply pgrx/src/datum/unbox.rs]: Ugh.

Comment (workingjubilee) [reply pgrx/src/datum/unbox.rs]: Can these have `#[rustfmt::skip]` instead?

Comment (hamirmahal) [reply pgrx/src/datum/unbox.rs]: Hmm... I'll take a look right now.

File: pgrx-sql-entity-graph/src/aggregate/mod.rs [hunk 2]
```diff
@@ -620,7 +616,7 @@ impl ToEntityGraphTokens for PgAggregate {
         let snake_case_target_ident =
             Ident::new(&target_ident.to_string().to_case(Case::Snake), target_ident.span());
         let sql_graph_entity_fn_name = syn::Ident::new(
-            &format!("__pgrx_internals_aggregate_{}", snake_case_target_ident),
+            &format!("__pgrx_internals_aggregate_{snake_case_target_ident}"),
             target_ident.span(),
```
Comment (workingjubilee) [inline pgrx-sql-entity-graph/src/aggregate/mod.rs]: All the cases where this `Ident::new` pattern is used should remain the same for reasons that are somewhat complex and have to do with "this pattern is suboptimal, but https://github.com/dtolnay/quote/issues/206 is too, and it's not just a formatting difference."

**Review by workingjubilee — CHANGES_REQUESTED** [review]

**Comment by workingjubilee** [comment]
Tests can be fixed with `TRYBUILD=overwrite`

**Comment by hamirmahal** [comment]
I'm getting some gnarly error messages. I followed the instructions at https://github.com/pgcentralfoundation/pgrx#hacking.

- `export PATH="$HOME/pgrx/bin:$PATH"` is in `~/.bashrc`.
- I have this repository on disk at `~/pgrx`.

```
TRYBUILD=overwrite cargo test --features pg14
```

starts out fine, but quickly deteriorates into

<details>
<summary>Error messages</summary>

```
error[E0603]: function `main` is private
   --> /home/hamir/pgrx/pgrx/src/lib.rs:319:29
    |
319 |         pub use pgrx_embed::main;
    |                             ^^^^ private function
    |
note: the function `main` is defined here
   --> /tmp/.tmp9wRy3Q:1:1
    |
1   | fn main () { pgrx_tests :: __pgrx_marker () ; let mut entities = Vec :: new () ; let contro...
    | ^^^^^^^^^^

error[E0601]: `main` function not found in crate `pgrx_embed_pgrx_tests`
 --> pgrx-tests/./src/bin/pgrx_embed.rs:1:23
  |
1 | ::pgrx::pgrx_embed!();
  | --------------------- ^ the main function must be defined at the crate level (in `pgrx-tests/./src/bin/pgrx_embed.rs`)
  | |
  | non-function item at `crate::main` is found
  |
note: here is a function named `main`
 --> /tmp/.tmp9wRy3Q:1:1
  |
1 | fn main () { pgrx_tests :: __pgrx_marker () ; let mut entities = Vec :: new () ; let control_...
  | ^^^^^^^^^^
  = note: you have one or more functions named `main` not defined at the crate level
  = help: consider moving the `main` function definitions

Some errors have detailed explanations: E0601, E0603.
For more information about an error, try `rustc --explain E0601`.
error: could not compile `pgrx-tests` (bin "pgrx_embed_pgrx-tests") due to 2 previous errors
test tests::aggregate_tests::tests::pg_aggregate_demo_custom_state ... FAILED
       Using PgConfig("pg14") and `pg_config` from /home/hamir/.pgrx/14.13/pgrx-install/bin/pg_config
    Building extension with features pg14 pg_test proptest
     Running command "/home/hamir/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo" "build" "--lib" "--features" "pg14 pg_test proptest" "--no-default-features" "--message-format=json-render-diagnostics"
warning: creating a shared reference to mutable static is discouraged
   --> pgrx-tests/src/tests/memcxt_tests.rs:112:28
    |
112 |                 assert_eq!(pg_sys::CurrentMemoryContext, ctx_sys);
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ shared reference to mutable static
    |
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html>
    = note: shared references to mutable statics are dangerous; it's undefined behavior if the static is mutated or if a mutable reference is created for it while the shared reference lives
    = note: `#[warn(static_mut_refs)]` on by default

warning: `pgrx-tests` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.10s
  Installing extension
     Copying control file to /home/hamir/.pgrx/14.13/pgrx-install/share/postgresql/extension/pgrx_tests.control
     Copying shared library to /home/hamir/.pgrx/14.13/pgrx-install/lib/postgresql/pgrx_tests.so
  Discovered 948 SQL entities: 58 schemas (6 unique), 832 functions, 22 types, 2 enums, 6 sqls, 1 ords, 1 hashes, 9 aggregates, 17 triggers
  Rebuilding for SQL generation with features `pg14 pg_test proptest`
warning: creating a shared reference to mutable static is discouraged
   --> pgrx-tests/src/tests/memcxt_tests.rs:112:28
    |
112 |                 assert_eq!(pg_sys::CurrentMemoryContext, ctx_sys);
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ shared reference to mutable static
    |
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html>
    = note: shared references to mutable statics are dangerous; it's undefined behavior if the static is mutated or if a mutable reference is created for it while the shared reference lives
    = note: `#[warn(static_mut_refs)]` on by default

warning: `pgrx-tests` (lib) generated 1 warning
   Compiling pgrx-tests v0.12.7 (/home/hamir/pgrx/pgrx-tests)
error[E0603]: function `main` is private
   --> /home/hamir/pgrx/pgrx/src/lib.rs:319:29
    |
319 |         pub use pgrx_embed::main;
    |                             ^^^^ private function
    |
note: the function `main` is defined here
   --> /tmp/.tmpkcVQsp:1:1
    |
1   | fn main () { pgrx_tests :: __pgrx_marker () ; let mut entities = Vec :: new () ; let contro...
    | ^^^^^^^^^^

error[E0601]: `main` function not found in crate `pgrx_embed_pgrx_tests`
 --> pgrx-tests/./src/bin/pgrx_embed.rs:1:23
  |
1 | ::pgrx::pgrx_embed!();
  | --------------------- ^ the main function must be defined at the crate level (in `pgrx-tests/./src/bin/pgrx_embed.rs`)
  | |
  | non-function item at `crate::main` is found
  |
note: here is a function named `main`
 --> /tmp/.tmpkcVQsp:1:1
  |
1 | fn main () { pgrx_tests :: __pgrx_marker () ; let mut entities = Vec :: new () ; let control_...
  | ^^^^^^^^^^
  = note: you have one or more functions named `main` not defined at the crate level
  = help: consider moving the `main` function definitions

Some errors have detailed explanations: E0601, E0603.
For more information about an error, try `rustc --explain E0601`.
error: could not compile `pgrx-tests` (bin "pgrx_embed_pgrx-tests") due to 2 previous errors
```
</details>

repeated a lot of times.

**Review by workingjubilee — APPROVED** [review]

**Pull request merged**